### PR TITLE
Fix wireprotocol for simulation

### PR DESCRIPTION
**Pull request by nikkolasg**
The new wire protocol design is not compatible for simulations because one simulation uses different "network" packages, one for each process, so there are packets which are non registered so it's not possible to Unmarshal them.

This PR fixes that by simply registering the packets a protocol use at init time.

### CODE REVIEW

File: network/encoding.go [hunk 1]
```diff
@@ -186,7 +185,7 @@ var marshalLock sync.Mutex
 
 // MarshalRegisteredType will marshal a struct with its respective type into a
 // slice of bytes. That slice of bytes can be then decoded in
-// UnmarshalRegisteredType.
+// UnmarshalRegisteredType. data MUST BE a pointer to the message.
```
Comment (ineiti) [inline network/encoding.go]: Please don't capitalize comments.

File: protocols/bftcosi/packets.go [hunk 2]
```diff
@@ -4,10 +4,24 @@ import (
...
+func init() {
+	for _, i := range []interface{}{
+		BFTSignature{},
+		Announce{},
+		Commitment{},
+		ChallengePrepare{},
+		ChallengeCommit{},
+		Response{},
+	} {
+		network.RegisterPacketType(i)
```
Comment (ineiti) [inline protocols/bftcosi/packets.go]: Should we add a `network.RegisterPacketTypes`? Like there are other `Register*s`-functions?

File: sda/overlay.go [hunk 3]
```diff
@@ -730,7 +730,7 @@ func (d *defaultProtoIO) Unwrap(msg interface{}) (interface{}, *OverlayMessage,
 	case ProtocolMsg:
 		sdaMsg := inner
 		var err error
-		_, msg, err := network.UnmarshalRegistered(sdaMsg.MsgSlice)
+		_, unmar, err := network.UnmarshalRegistered(sdaMsg.MsgSlice)
```
Comment (ineiti) [inline sda/overlay.go]: I prefer `msg` over `unmar`...

Comment (nikkolasg) [inline sda/overlay.go]: I've named it `protoMsg` then, is it ok? I don't like shadowing a variable and `msg` is already the name of an argument.

File: protocols/bftcosi/packets.go (hunk 2 above)
Comment (nikkolasg) [inline protocols/bftcosi/packets.go]: Yes I agree it'd useful. But on the other hand we already have many ways to register a packet (with pointers, with uuids, and decode returns pointer or values etc) and I'd prefer to make an issue about `Cleaning network.RegisterXXX methods` and adding this new method inside the issue. 
Is it ok with you ?

Comment (ineiti) [inline protocols/bftcosi/packets.go]: OK, open the issue and I'll close the comments.

Comment (nikkolasg) [inline protocols/bftcosi/packets.go]: here it is: #722

**Pull request merged**